codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttMessage.java

```diff
@@ -61,7 +54,7 @@ public Object variableHeader() {
     }
 
     public Object payload() {
```

Comment:
@Scottmitch you are right in terms of back compatibility. However this field **should** return null for messages that have no payload. This is logical from point of view of mqtt spec.
I can return removed constructor and mark it deprecated. But from point of view this field should return null for messages with no payload.
Let me know if you are ok with it, otherwise please close this PR.